File: julia/multithread.jl
import Base.Threads.@spawn


function sub_func()::Float64
    values::Array{Float64, 1} = randn(Float64, 20000000)
    total_values = sum(values)
end


function main()
    tasks = []
    for i = 1:500
        t = @spawn sub_func()
        push!(tasks, t)
    end
    for t in tasks
        wait(t)
    end
end

@time main()

File: python/multithread.py
from concurrent.futures import ProcessPoolExecutor
import numpy as np


def sub_func():
    values = np.random.randn(20000000)
    total_values = np.sum(values)


def main():
    tasks = []
    with ProcessPoolExecutor(max_workers=16) as pool:
        for i in range(1, 501):
            t = pool.submit(sub_func)
            tasks.append(t)

    for t in tasks:
        t.result()


if __name__ == "__main__":
    import datetime as dt
    start = dt.datetime.now()
    main()
    print(dt.datetime.now() - start)
